linkerd/app/inbound/src/policy/discover.rs

```diff
@@ -176,32 +177,60 @@ fn to_policy(proto: api::Server) -> Result<ServerPolicy> {
                     authn => return Err(format!("no authentication provided: {:?}", authn).into()),
                 };
 
-                let name = labels
-                    .get("name")
-                    .ok_or("authorization missing 'name' label")?
-                    .clone()
-                    .into();
+                let (kind, name) = {
+                    let name = labels
+                        .get("name")
+                        .ok_or("authorization missing 'name' label")?
+                        .clone();
+
+                    let mut parts = name.splitn(2, ':');
+                    match (parts.next().unwrap(), parts.next()) {
+                        (kind, Some(name)) => (kind.into(), name.into()),
+                        (name, None) => {
+                            let kind = labels
+                                .get("kind")
+                                .cloned()
+                                .unwrap_or_else(|| "serverauthorization".to_string());
+                            (kind.into(), name.into())
+                        }
+                    }
```

Comment:
nit/TIOLI: this looks like a parsing helper that can be factored out? not a big deal though, as it's only repeated twice